hy/compiler.py

```diff
@@ -1134,12 +1133,16 @@ def compile_unary_operator(self, expr, root, arg):
         return operand + asty.UnaryOp(
             expr, op=ops[root](), operand=operand.force_expr)
 
-    _symn = some(lambda x: isinstance(x, HySymbol) and "." not in x)
+    def _importlike(*name_types):
+        name = some(lambda x: isinstance(x, name_types) and "." not in x)
```

Comment:
```diff
-        name = some(lambda x: isinstance(x, name_types) and "." not in x)
+        name = some(lambda x: isinstance(x, name_types) and (x.startswith("#") or "." not in x))
```
otherwise you can't import e.g. `"#."`

Reply:
But if a macro's name has a period in it, how do you distinguish that kind of period from the scoping kind of period?

Reply:
Ah, I didn't think tag macros could be scoped.
Would we want to have tag macros like `#module.tagname`?

Reply:
Yes, requiring a tag macro in a namespace-qualified fashion should work just as with ordinary macros, with or without this PR.

Reply:
If you can't import `"#."` shouldn't it be illegal to define a macro with that name too? Currently you can `(defmacro "#." [a])` but it's only usable in that module which doesn't seem right. Also should apply to a regular `(defmacro . [a])` since `.` is a special form and shadowing it with a macro kind of breaks everything + isn't even possible when using `defn` anyways.

Reply:
It seems there's a variety of interrelated, preexisting bugs involving `require` on names that get mangled and names that get required in a fully qualified fashion. I'm going to ban periods in `defmacro`, as you suggest, and leave those additional bugs for later PRs.